emukit/bayesian_optimization/acquisitions/constrained_bayesian_optimization.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+
+import numpy as np
+
+from ...bayesian_optimization.acquisitions import ExpectedImprovement, ProbabilityOfFeasibility
+from ...core.acquisition import Acquisition
+from ...core.interfaces import IModel
+from ...core.loop import FixedIntervalUpdater, OuterLoop, SequentialPointCalculator
+from ...core.loop.loop_state import create_loop_state
+from ...core.optimization import AcquisitionOptimizer
+from ...core.parameter_space import ParameterSpace
+
+
+class ConstrainedBayesianOptimizationLoop(OuterLoop):
```

Comment:
Can we move this to `emukit.bayesian_optimization.loops`? @apaleyes do you agree? Or should it go in examples?

Reply:
it seems generic enough to go into emukit.bayesian_optimization.loops?

Reply:
Absolutely. I am just not sure about the name. Is it too generic for what this loop does? How would one call bayes opt loop when constraints are known, like in #212?

Reply:
What about UnknownConstraintsBayesianOptimizationLoop?